src/c++/perf_analyzer/genai-perf/genai_perf/logging.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+import logging.config
+from pathlib import Path
+
+from genai_perf.constants import DEFAULT_DATE_FORMAT, DEFAULT_LOG_FORMAT
+
+
+def init_logging(log_file: Path) -> None:
+    LOGGING_CONFIG = {
+        "version": 1,
+        "disable_existing_loggers": False,
+        "formatters": {
+            "standard": {
+                "format": DEFAULT_LOG_FORMAT,
+                "datefmt": DEFAULT_DATE_FORMAT,
+            },
+        },
+        "handlers": {
+            "console": {
+                "level": "INFO",
+                "formatter": "standard",
+                "class": "logging.StreamHandler",
+                "stream": "ext://sys.stdout",  # Default is stderr
+            },
+            "file": {
+                "level": "WARNING",
+                "formatter": "standard",
+                "class": "logging.FileHandler",
```

Comment:
Is there a place in the code where an error may be thrown if the file exists or cannot be modified? If so, do we want to catch it?

Reply:
Thanks for explaining. What I mean was what if an error is thrown in the instantiation of the logger? Like the file cannot be opened?

Reply:
I would assume it would just throw an exception and the program would crash.

Reply:
Do you want to leave it as is? I would think that's something we'd want to catch and handle gracefully.

CC: @rmccorm4 who might know better what we do for Python logging in general.

Reply:
isn't this moot if we removed `--log-file`?

Reply:
Good point. There might still be a risk if there is a permissions error on the user's side, but that case is probably rare enough it's not worth covering for now. I think the Python backend had a misleading error at some point due to the user's system not having permissions to write the correct file, but we can handle that when/if it happens.

Reply:
I have an AI to create an option to pass a path for all artifacts so the user can ensure they have permission to generate any files in the first place.
We removed the log file here but the core issue is still worth noting. If a user is using our tool but they do not have permission to store something to a directory, our program will blow up.
Its a good edge case to think about and it will be addressed.